hal/src/rtl872x/interrupts_hal.cpp

```diff
@@ -87,6 +87,64 @@ void gpioIntHandler(void* data) {
     }
 }
 
+// Particle implementation of GPIO_INTMode (without the PAD_PullCtrl() call)
+void GPIO_INTMode_HAL(u32 GPIO_Pin, u32 GPIO_Port, u32 GPIO_ITEnable, u32 GPIO_ITTrigger, u32 GPIO_ITPolarity, u32 GPIO_ITDebounce) {
+    uint32_t pinMask = 1 << (GPIO_Pin & 0x1f);
+    uint32_t portBase = (GPIO_Port == RTL_PORT_A) ? (uint32_t)GPIOA_BASE : (uint32_t)GPIOB_BASE;
+    uint32_t tempReg = 0;
+
+    // Explicitly do NOT set PULL UP/PULL DOWN PADCTL register
+
+    // SET GPIO_INT_BOTHEDGE reg
+    tempReg = HAL_READ32(portBase, 0x68);
+    if (GPIO_ITTrigger == GPIO_INT_Trigger_BOTHEDGE) {
+        HAL_WRITE32(portBase, 0x68, tempReg | pinMask);
+    } else {
+        HAL_WRITE32(portBase, 0x68, tempReg & ~pinMask);
+        // SET GPIO_INTTYPE_LEVEL reg only if not BOTH EDGE
+        tempReg = HAL_READ32(portBase, 0x38);
+        if (GPIO_ITTrigger == GPIO_INT_Trigger_EDGE) {
+            HAL_WRITE32(portBase, 0x38, tempReg | pinMask);
+        } else {
+            HAL_WRITE32(portBase, 0x38, tempReg & ~pinMask);
+        }
+    }
+    // SET GPIO_INT_POLARITY reg
+    tempReg = HAL_READ32(portBase, 0x3c);
+    if (GPIO_ITPolarity == GPIO_INT_POLARITY_ACTIVE_HIGH) {
+        HAL_WRITE32(portBase, 0x3c, tempReg | pinMask);
+    } else {
+        HAL_WRITE32(portBase, 0x3c, tempReg & ~pinMask);
+    }
+    // SET GPIO_DEBOUNCE reg
+    tempReg = HAL_READ32(portBase, 0x48);
+    if (GPIO_ITDebounce == GPIO_INT_DEBOUNCE_ENABLE) {
+        HAL_WRITE32(portBase, 0x48, tempReg | pinMask);
+    } else {
+        HAL_WRITE32(portBase, 0x48, tempReg & ~pinMask);
+    }
+    // SET GPIO_INTEN reg
+    tempReg = HAL_READ32(portBase, 0x30);
+    if (GPIO_ITEnable == ENABLE) {
+        HAL_WRITE32(portBase, 0x30, tempReg | pinMask);
+    } else {
+        HAL_WRITE32(portBase, 0x30, tempReg & ~pinMask);
+    }
+}
+
+// Particle implementation of GPIO_Init (which calls our GPIO_INTMode_HAL to avoid PAD_PullCtrl() call)
+// NOTE: Only for use with GPIO_Mode_INT, do not use for GPIO_Mode_IN/GPIO_Mode_OUT
+// NOTE: hal_pin_is_valid(pin) must be called prior to this!
+void GPIO_Init_HAL(GPIO_InitTypeDef *GPIO_InitStruct, u32 GPIO_Port) {
+    if (GPIO_InitStruct->GPIO_Mode != GPIO_Mode_INT) {
+        return;
+    }
+
+    Pinmux_Config(GPIO_InitStruct->GPIO_Pin, GPIO_Mode_IN); // force INPUT mode
```

Comment:
This doesn't look right. `Pinmux_Config` configures the pin function, pin functions are:
```
#define PINMUX_FUNCTION_GPIO				(0)
#define PINMUX_FUNCTION_UART				(1)

#define PINMUX_FUNCTION_UART_RTSCTS		(2)
#define PINMUX_FUNCTION_LOGUART			(2)

#define PINMUX_FUNCTION_SPIM				(3)
#define PINMUX_FUNCTION_SPIS				(3)

#define PINMUX_FUNCTION_RTC				(4)
#define PINMUX_FUNCTION_TIMINPUT			(4)
#define PINMUX_FUNCTION_EXT32K			(28)
#define PINMUX_FUNCTION_RTCOUT			(28)
#define PINMUX_FUNCTION_TIMINPUT_HS		(22)

#define PINMUX_FUNCTION_IR					(5)
#define PINMUX_FUNCTION_SPIF				(6)
#define PINMUX_FUNCTION_I2C				(7)

#define PINMUX_FUNCTION_SDIOD				(8)
#define PINMUX_FUNCTION_SDIOH				(8)

#define PINMUX_FUNCTION_PWM				(9)
#define PINMUX_FUNCTION_PWM_HS			(9)
#define PINMUX_FUNCTION_PWM_LP			(10)
#define PINMUX_FUNCTION_SWD				(11)

#define PINMUX_FUNCTION_I2S				(12)
#define PINMUX_FUNCTION_DMIC				(12)

#define PINMUX_FUNCTION_KEYSCAN_ROW		(29)
#define PINMUX_FUNCTION_KEYSCAN_COL		(30)PAD_BIT_MASK_FUNCTION_ID
#define PINMUX_FUNCTION_LCD				(13)
#define PINMUX_FUNCTION_USB				(14)
#define PINMUX_FUNCTION_QDEC				(15)
#define PINMUX_FUNCTION_SGPIO				(16)
#define PINMUX_FUNCTION_RFE				(18)
#define PINMUX_FUNCTION_BTCOEX			(19)

#define PINMUX_FUNCTION_WIFIFW			(20)
#define PINMUX_FUNCTION_EXT_PCM			(20)
#define PINMUX_FUNCTION_BB_PIN			(21)
#define PINMUX_FUNCTION_SIC				(22)
#define PINMUX_FUNCTION_WAKEUP			(31)
#define PINMUX_FUNCTION_DBGPORT			(23)
#define PINMUX_FUNCTION_BBDBG				(25)
```

Reply:
Good catch!  I had documented this, but missed it in the implementation.  They are both 0 though so it still worked :D